Api/RequestHandler.php

```diff
@@ -196,12 +200,16 @@ protected function decryptLast4( $payment )
         if (strlen($last4) == 4 && ctype_digit($last4)) {
             return $last4;
         }
+
+        if ( strlen($last4) > 4 || empty($last4) ){
+            return self::DEFAULT_LAST_4; //spoof variable so things get filled in.
+        }
     }
 
     protected function formatCcType( $code )
```

Comment:
If were running on PHP 7.0 might be good to add a type hint here `protected function formatCcType(string $code)` this should give you better editor support and some extra type-safety.